core/db.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * @module core/db
+ */
+
+'use strict';
+
+const path        = require('path');
+const util        = require('util');
+const stream      = require('stream');
+const levelup     = require('levelup');
+const leveldown   = require('leveldown');
+const Genesis     = require('lib/genesis');
+const genesis     = require(process.env.GENESIS_PATH || 'genesis.json');
+
+/**
+ * Data directory for client
+ *
+ * ENV: datadir
+ * @default 'data'
+ * @type {String}
+ */
+const DATADIR = process.env.DATADIR || 'data';
+
+const CHAIN  = 'chain';
+const POOL   = 'pool';
+
+/**
+ * @desc Base class for DataBases
+ */
+class DB {
+
+    /**
+     * @param {String} name Name of the database (also name of the folder in DATADIR)
+     */
+    constructor(name) {
+        const db           = levelup(leveldown(path.join(DATADIR, name)));
+        const genesisBlock = Genesis.genesisToBlock(genesis);
+
+        Object.defineProperty(this, 'name', {value: name});
+        Object.defineProperty(this, 'db',   {value: db});
+        Object.defineProperty(this, 'genesis', {value: genesisBlock});
+    }
+
+    /**
+     * Get values stream
+     *
+     * @return {stream.Readable}
+     */
+    createReadStream() {
+        return this.db.createValueStream();
+    }
+
+    /**
+     * Destroy database by it's name
+     *
+     * @param  {String}  name Name of the database to destroy
+     * @return {Promise}      Promise that's resolved on destruction end
+     */
+    destroy() {
+        return this.db
+            .close()
+            .then(() => util.promisify(leveldown.destroy.bind(leveldown))(path.join(DATADIR, this.name)))
+            .then(() => this.db.open());
+    }
+}
+
+/**
+ * @desc Pool logic class
+ */
+class Pool extends DB {
+
+    /**
+     * Add transaction to the pool
+     *
+     * @param {String} tx Transaction to store
+     */
+    add(tx) {
+        return this.db.put(tx, tx);
```

Comment:
why not use transaction hash as key?

Reply:
1) There's no access to TX key in this part of app and it's not a DB logic to extract it.
2) Currently there's no reason to do it as pool is always drained and no BP would access pool's transactions by tx hash.